src/NuGet.Core/NuGet.Build.Tasks/GetRestoreDotnetCliToolsTask.cs

```diff
@@ -103,7 +103,7 @@ public override bool Execute()
                 BuildTasksUtility.CopyPropertyIfExists(msbuildItem, properties, "Version");
 
                 properties.TryGetValue("Version", out string value);
-                var uniqueName = ToolRestoreUtility.GetUniqueName(msbuildItem.ItemSpec, ToolFramework, VersionRange.Parse(value));
+                var uniqueName = ToolRestoreUtility.GetUniqueName(msbuildItem.ItemSpec, ToolFramework, value != null ? VersionRange.Parse(value) : null);
```

Comment:
any reason not to use VersionRange.All? does that get printed out in a strange way?